llvm/lib/Target/LoongArch/MCTargetDesc/LoongArchAsmBackend.h

```diff
@@ -75,6 +87,9 @@ class LoongArchAsmBackend : public MCAsmBackend {
   std::unique_ptr<MCObjectTargetWriter>
   createObjectTargetWriter() const override;
   const MCTargetOptions &getTargetOptions() const { return TargetOptions; }
+  DenseMap<MCSection *, const MCSymbolRefExpr *> &getSecToAlignSym() {
```

Comment:
This does not match binutils. Binutils does not generate `.Lla-relax-align` for each section.
```
$ cat a.s
.section .text2, "ax"
.p2align 4

.text
.p2align 4
```

- readelf -rSs
```
There are 10 section headers, starting at offset 0x178:

Section Headers:
  [Nr] Name              Type             Address           Offset
       Size              EntSize          Flags  Link  Info  Align
  [ 0]                   NULL             0000000000000000  00000000
       0000000000000000  0000000000000000           0     0     0
  [ 1] .text             PROGBITS         0000000000000000  00000040
       000000000000000c  0000000000000000  AX       0     0     16
  [ 2] .rela.text        RELA             0000000000000000  00000108
       0000000000000018  0000000000000018   I       7     1     8
  [ 3] .data             PROGBITS         0000000000000000  0000004c
       0000000000000000  0000000000000000  WA       0     0     1
  [ 4] .bss              NOBITS           0000000000000000  0000004c
       0000000000000000  0000000000000000  WA       0     0     1
  [ 5] .text2            PROGBITS         0000000000000000  00000050
       000000000000000c  0000000000000000  AX       0     0     16
  [ 6] .rela.text2       RELA             0000000000000000  00000120
       0000000000000018  0000000000000018   I       7     5     8
  [ 7] .symtab           SYMTAB           0000000000000000  00000060
       0000000000000090  0000000000000018           8     6     8
  [ 8] .strtab           STRTAB           0000000000000000  000000f0
       0000000000000012  0000000000000000           0     0     1
  [ 9] .shstrtab         STRTAB           0000000000000000  00000138
       000000000000003d  0000000000000000           0     0     1
Key to Flags:
  W (write), A (alloc), X (execute), M (merge), S (strings), I (info),
  L (link order), O (extra OS processing required), G (group), T (TLS),
  C (compressed), x (unknown), o (OS specific), E (exclude),
  D (mbind), p (processor specific)

Relocation section '.rela.text' at offset 0x108 contains 1 entry:
  Offset          Info           Type           Sym. Value    Sym. Name + Addend
000000000000  000500000066 R_LARCH_ALIGN     0000000000000000 .Lla-relax-align + 4

Relocation section '.rela.text2' at offset 0x120 contains 1 entry:
  Offset          Info           Type           Sym. Value    Sym. Name + Addend
000000000000  000500000066 R_LARCH_ALIGN     0000000000000000 .Lla-relax-align + 4

Symbol table '.symtab' contains 6 entries:
   Num:    Value          Size Type    Bind   Vis      Ndx Name
     0: 0000000000000000     0 NOTYPE  LOCAL  DEFAULT  UND 
     1: 0000000000000000     0 SECTION LOCAL  DEFAULT    1 .text
     2: 0000000000000000     0 SECTION LOCAL  DEFAULT    3 .data
     3: 0000000000000000     0 SECTION LOCAL  DEFAULT    4 .bss
     4: 0000000000000000     0 SECTION LOCAL  DEFAULT    5 .text2
     5: 0000000000000000     0 NOTYPE  LOCAL  DEFAULT    5 .Lla-relax-align
```

Reply:
So I think there may be some potential errors if the section of the symbol ".Lla-relax-align" is inconsistent with the section of the alignment directive. And I submitted patch before (see [1]). For example if the section of the ".Lla-relax-align" is discard. And another example for linux kernel [2].

[1] https://sourceware.org/pipermail/binutils/2024-January/131615.html
[2] https://lore.kernel.org/loongarch/2abbb633-a10e-71cc-a5e1-4d9e39074066@loongson.cn/T/#t